pkg/finality-grandpa/vote_graph.go

```diff
@@ -0,0 +1,699 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"fmt"
+
+	"github.com/tidwall/btree"
+	"golang.org/x/exp/constraints"
+	"golang.org/x/exp/slices"
+)
+
+type voteGraphEntry[
+	Hash constraints.Ordered,
```

Comment:
Why do hashes need to be constrained to be ordered? Not clear to me how ordering of hashes works here

Reply:
They need to be ordered when searching through hashes.